lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1749,6 +1749,107 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
+                .toBuiltinTensor();
+        SmallVector<int64_t> shapeSizes(tensorType.getShape());
+        for (unsigned i = 0; i < shapeSizes.size(); i++) {
+          if (shapeSizes[i] == ShapedType::kDynamic)
+            shapeSizes[i] = Torch::kUnknownSize;
+        }
+
+        Value cst0 = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(0));
+        Value cst1 = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(1));
+        Value nVal = rewriter.create<Torch::AtenSizeIntOp>(binder.getLoc(),
+                                                           operand, cst0);
+        Value mVal = rewriter.create<Torch::AtenSizeIntOp>(binder.getLoc(),
+                                                           operand, cst1);
+        Value noneVal = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
+        if (dtypeIntOnnx == -1) {
+          dtypeVal = noneVal;
+        } else {
+          dtypeIntTorch = onnxDtypeIntToTorchDtypeInt(dtypeIntOnnx);
+          dtypeVal = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(dtypeIntTorch));
+        }
+
+        if (diagonalIndex == 0) {
+          // diagonalIndex = 0 populates the main diagonal
+          rewriter.replaceOpWithNewOp<Torch::AtenEyeMOp>(
+              binder.op, resultType, nVal, mVal, dtypeVal, noneVal, noneVal,
+              noneVal);
+          return success();
+        } else {
+          // diagonalIndex > 0 populates an upper diagonal
+          // diagonalIndex < 0 populates a lower diagonal
+          // create a main diag eye op, then embed into a zeros op
+          Value diagVal = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(),
+              rewriter.getI64IntegerAttr(std::abs(diagonalIndex)));
+          SmallVector<int64_t> eyeShape = shapeSizes;
+          Value newN, newM, dimVal, startVal;
+
+          // get shapes of main diag eye op and zeros op
+          if (diagonalIndex > 0) {
+            newN = nVal;
+            newM = rewriter.create<Torch::AtenSubIntOp>(binder.getLoc(), mVal,
+                                                        diagVal);
+            if (eyeShape[1] != Torch::kUnknownSize) {
+              eyeShape[1] -= diagonalIndex;
+            }
+            dimVal = cst1;
+            startVal = mVal;
+          } else {
+            newN = rewriter.create<Torch::AtenSubIntOp>(binder.getLoc(), nVal,
+                                                        diagVal);
+            newM = mVal;
+            if (eyeShape[0] != Torch::kUnknownSize) {
+              eyeShape[0] += diagonalIndex;
+            }
+            dimVal = cst0;
+            startVal = nVal;
+          }
+
+          // create main diag eye op
+          auto eyeResultType =
+              Torch::ValueTensorType::get(binder.op->getContext(), eyeShape,
```

Comment:
You can do `rewriter.getType<Torch::ValueTensorTyoe>(...)` so that you do not need to request the context.